src/main.rs

```diff
@@ -0,0 +1,116 @@
+use std::{
+    env, fs,
+    io::{stderr, Write},
+    path::Path,
+    process,
+};
+
+use getopts::Options;
+use libc::{c_int, getchar, putchar};
+
+fn usage(prog: &str) -> ! {
+    let path = Path::new(prog);
```

Comment:
If I understand, all of this is just to get the basename of argv[0]?

Reply:
Yes.